docs/doc/60-contributing/03-rfcs/20221017-multiple-catalog.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Multiple Catalog
+description: multiple catalog support for databend
+---
+
+- RFC PR: [datafuselabs/databend#0000](https://github.com/datafuselabs/databend/pull/0000)
+- Tracking Issue: [datafuselabs/databend#0000](https://github.com/datafuselabs/databend/issues/0000)
+
+## Summary
+
+Allow users to maintain multiple catalogs for the databend.
+
+## Motivation
+
+Databend supports multiple catalogs now, but only in a static way.
+
+To allow accessing the `hive` catalog, users need to configure `hive` inside `databend-query.toml` in this way:
+
+```toml
+[catalog]
+meta_store_address = "127.0.0.1:9083"
+protocol = "binary"
+```
+
+Users can't add/alter/remove the catalogs during runtime.
+
+By allowing users to maintain multiple catalogs for the databend, we can integrate more catalogs like `iceberg` more quickly.
+
+## Guide-level explanation
+
+After this RFC has been implemented, users can create new catalogs like:
+
+```sql
+CREATE CATALOG my_hive
+  TYPE='hive'
+  CONNECTION = (URL='<hive-meta-store>' THRIFT_PROTOCOL=BINARY);
+SELECT * FROM my_hive.DB.table;
+```
+
+Besides, users can alter or drop a catalog:
+
+```sql
+DROP CATALOG my_hive;
+```
+
+Users can add more catalogs like:
+
+```sql
+CREATE CATALOG my_iceberg
+  TYPE='iceberg'
+  CONNECTION = (URL='s3://my_bucket/path/to/iceberg');
+SELECT * FROM my_iceberg.DB.table;
+```
+
+With this feature, users can join data from different catalogs now:
+
+```sql
+select
+    my_iceberg.DB.purchase_records.Client_ID,
+    my_iceberg.DB.purchase_records.Item,
+    my_iceberg.DB.purchase_records.QTY
+from my_hive.DB.vip_info
+inner join my_iceberg.DB.purchase_records
+    on my_hive.DB.vip_info.Client_ID = my_iceberg.DB.purchase_records.Client_ID;
+```
+
+## Reference-level explanation
+
+Databend has a framework for multiple catalogs now. The only change for us is to store catalog-related information in metasrv instead.
+
+To make it possible to start a query without `metasrv`, we will also support configuring catalogs in config like:
+
+```toml
+[catalogs.my_hive]
+meta_store_address = "127.0.0.1:9083"
+protocol = "binary"
+
+[catalogs.my_iceberg]
+URL = "s3://bucket"
+```
```

Comment:
databend-query starts with either a remote meta-service or an embedded one.

Do we have to store catalog info in a config file?

Reply:
Store catalog in config so that users who only want to visit `hive` or `iceberg` don't need to access metasrv. It also makes it easier for existing hive users to migrate.